lib/graphql/schema/printer.rb

```diff
@@ -0,0 +1,131 @@
+module GraphQL
+  # Used to convert your {GraphQL::Schema} to a GraphQL schema string
+  #
+  # @example print your schema to standard output
+  #   Schema = GraphQL::Schema.new(query: QueryType)
+  #   puts GraphQL::Schema::Printer.print_schema(Schema)
+  #
+  module Schema::Printer
+    extend self
+
+    # Return a GraphQL schema string for the defined types in the schema
+    # @param schema [GraphQL::Schema]
+    def print_schema(schema)
+      print_filtered_schema(schema, method(:is_defined_type))
+    end
+
+    # Return the GraphQL schema string for the introspection type system
+    def print_introspection_schema
+      query_root = ObjectType.define do
+        name "Query"
+      end
+      schema = Schema.new(query: query_root)
+      print_filtered_schema(schema, method(:is_introspection_type))
+    end
+
+    private
+
+    def print_filtered_schema(schema, type_filter)
+      types = schema.types.values.select{ |type| type_filter.call(type) }.sort_by(&:name)
+      types.map{ |type| print_type(type) }.join("\n\n")
+    end
+
+    BUILTIN_SCALARS = Set.new(["String", "Boolean", "Int", "Float", "ID"])
+    private_constant :BUILTIN_SCALARS
+
+    def is_introspection_type(type)
+      type.name.start_with?("__")
+    end
+
+    def is_defined_type(type)
+      !is_introspection_type(type) && !BUILTIN_SCALARS.include?(type.name)
+    end
+
+    def print_type(type)
+      case type
```

Comment:
I pushed commit 0b5f05fea51b7da3d41434c2883a2d6f7b58dbd3 which takes the approach you suggested.  The only disadvantage would be that it is more verbose, but I have nothing against it if it makes the code more readable.